Design fixes

DanielHeath opened the pull request.

DanielHeath committed:
Fix size of 'done' button per design feedback

DanielHeath committed:
Set displayName for components which need it

DanielHeath committed:
Only pass what we need TransportTime

DanielHeath committed:
Fix line-height for ‘Other contact options’

danni commented:
You've lost the padding/line-height of the Other contact details link.

pfctdayelise committed:
Add isConfidential attribute to Service for better readability

DanielHeath committed:
Add utility class in new `styles/shared` folder (for mixins etc)

DanielHeath committed:
Extract google maps link into its own component & Location class

DanielHeath committed:
Show transport time on both ServicePane and ResultList

DanielHeath committed:
Run visual tests

DanielHeath committed:
Use unique class for travel time to unconfuse the tests

danni commented on src/components/Address.js:
```diff
@@ -1,15 +1,15 @@
     // flow:disable not supported yet
     static sampleProps = {
-        complex: {
+        complex: {location: new Location({
```
newline

danni commented on src/components/Address.js:
```diff
@@ -20,60 +20,31 @@ class Address extends React.Component {
-
-        streetParts = _(streetParts)
-            .compact()
-            .map(part => `${part}, `)
```
This is just `.join(', ')`

DanielHeath replied:
It adds it to the last one as well, and doesn't add it if there are none.

DanielHeath committed:
Break out newline for clarity

danni commented on src/iss/Location.js:
```diff
@@ -0,0 +1,50 @@
+        addrDescriptors = _(addrDescriptors)
+            .compact()
+            .map(part => `${part}, `)
+            .join("");
```
`.join(', ')`

DanielHeath replied:
That wouldn't be the same thing.

DanielHeath replied:
Specifically, `[]` should result in `""` and `['Level 1']` should result in `"Level 1 "`

danni replied:
If you want a comma between `addrDescriptors` and `street` just finish with `[a, b].compact().join(', ')`

DanielHeath replied:
Not sure that's cleaner - I'll have a look for a good alternative.

danni replied:
I would also accept `join(', ') + ', '`

danni commented on src/pages/ResultsListPage.js:
```diff
@@ -149,7 +149,7 @@ class ResultsListPage extends BaseResultsPage {
                       />;
 
-                    var klass = elem.displayName || "other";
+                    var klass = elem.type.displayName || "other";
```
?

DanielHeath replied:
`elem.displayName` never worked - it always just went to `"other"`, both in dev and prod.

danni replied:
Okay so this is canonically correct now?

danni commented:
This is probably going to collide with @pfctdayelise who I think was handling showing the phrase 'Confidential location' if appropriate. Can you make sure that won't break/coordinate a merge.

pfctdayelise committed:
Add test for TravelTime info on service details page

pfctdayelise commented on src/components/ServicePane.js:
```diff
@@ -58,9 +60,16 @@ export default class ServicePane extends React.Component {
                     <hr />
                     <ContactMethods object={object} />
+                    <hr />
+                    <GoogleMapsLink
```
Note in mine I was linking to the Google Maps directions (starting point "Current Location")

DanielHeath replied:
ooh, good spot

DanielHeath committed:
Use Location object in TransportTime

pfctdayelise commented on src/components/TransportTime.js:
```diff
@@ -9,39 +9,70 @@ import icons from "../icons";
+            <div
+                className={`TransportTime ${this.compactClass}`}
+            >
+                <icons.Walk className="ColoredIcon" />
```
lost class brand-text-dark? (not that I am sure it does anything for icons...)

DanielHeath replied:
Right - I realized that the class wasn't defined in our CSS. Should be a separate commit though.

pfctdayelise commented on src/components/TransportTime.scss:
```diff
@@ -1,18 +1,24 @@
-    .time {
+    .travel-time {
         color: $brand-text-dark;
         font-weight: bold;
```
I asked Danni, this should actually be 500

DanielHeath committed:
Fixup: Make ‘Get directions’ case match the design

DanielHeath committed:
Fixup: Use correct designed font-weight

DanielHeath committed:
Move padding into component, controlled by `compact` prop

pfctdayelise commented:
In terms of the transport time info it looks to me this covers the same things that my change did. :+1:

DanielHeath committed:
Move google maps stuff into google maps component.

DanielHeath committed:
Fix tests for google maps link.

Matching link text through an <a> containing block elements is not straightforwards.

DanielHeath committed:
Alphabetize package.json

danni commented on src/components/Address.js:
```diff
@@ -1,79 +1,62 @@
-                    aria-label="Open Google Maps in a new tab"
-                    href={`https://maps.google.com/?q=${query}`}
-                >
+                <GoogleMapsLink to={location}>
```
I thought the Google Maps directions link was meant to go on the `TransportTime` component? Will accept a FIXME comment for this.

danni commented on src/components/GoogleMapsLink.js:
```diff
@@ -0,0 +1,67 @@
+
+    googleMapsUrl(): string {
+        var toAddr = this.props.to;
+        var start = "Current+Location";
```
Would be better as a 'from' property with a default of 'Current Location'.

DanielHeath committed:
Add fixme comments

DanielHeath commented:
Merging so Isaac can look at it.
